app/controllers/admin/api/services/proxy/policies_controller.rb

```diff
@@ -41,7 +40,7 @@ def show
   #
   def update
     if proxy.update_attributes(proxy_params)
-      proxy.deploy_v2 if proxy.apicast_configuration_driven
+      ProxyDeploymentService.new(proxy).deploy_v2 if proxy.apicast_configuration_driven
```

Comment:
I would not have 2 methods for a Service, only one `#call`. It should handle all the logic inside. It should act as a blackbox. This way you do not leak any logic outside.

Reply:
How about `deploy` and `deploy_production`? They don't depend on any variables (I think), would we need 2 services? `ProxyDeployment` and `ProxyDeploymentProduction`.. Or would it be better to pass an argument? `ProxyDeploymentService.call(@proxy, :production)`?

Reply:
Well maybe just do 

```ruby
class ProxyDeploymentService
  def initialize(proxy, stage: :staging)
       @stage = stage
       @proxy = proxy
  end

  def call
    case @stage
    when :staging, :sandbox
    when :production 
    end
  end
  end
end
```